paasta_tools/cli/cmds/spark_run.py

```diff
@@ -480,10 +490,46 @@ def get_spark_conf_str(
         )
         sys.exit(1)
 
-    spark_conf = list()
-    for opt, val in dict(non_user_args, **user_args).items():
-        spark_conf.append(f'--conf {opt}={val}')
-    return ' '.join(spark_conf)
+    return dict(non_user_args, **user_args)
+
+
+def create_spark_config_str(spark_config_dict):
+    spark_config_entries = list()
+    for opt, val in spark_config_dict.items():
+        spark_config_entries.append(f'--conf {opt}={val}')
+    return ' '.join(spark_config_entries)
+
+
+def emit_resource_requirements(spark_config_dict, paasta_cluster):
```

Comment:
Do we need to deal with exceptions here, e.g. writing to aws failure? I think spark-run can proceed with a message that cluster may not scale up?

Reply:
I'm a little unsure what the right move is.  In cases where it's a user using spark-run interactively, displaying a message and continuing may be fine -- maybe the user decides to cancel the run, maybe they let it hobble along.  

If it's being run as a schedule batch, is it still ok?  The batch may _never_ be able to get the resources it needs.  It might be better in that case to just fail and let tron (or whatever) retry if desired.  It also seems preferable to take the same approach in both cases (since otherwise we'd need some way to determine whether we're being run interactively or what).

I think is situation is a little trickier because of the way we're currently thinking about the Spark signal.  We aren't planning on running a "current resources" signal, so if a spark run fails to notify about its resources, there's no mechanism trying to make sure the "orphaned" run can keep its resources, let alone get more resources.

Seems like the options are:
1. Fail and let the user/scheduler retry if desired
2. Warn and go ahead, keeping the current signal implementation.  The job may never be able to scale up all the way (seems like it will, at best, be able to use the cluster
3. Warn and go ahead, but use a current resources signal and change the spark signal so that its job is only to report additional resources needed by spark jobs as they scale up
4. Warn and go ahead, and modify the spark signal so that it looks for orphaned spark jobs (jobs whose framework name looks like a spark job but have no resource request metrics) and reports their resources, more or less making it act as a current resources signal for _only_ oprhaned spark jobs.

I think 1 is the best for now and that we may want to consider moving to 3 if it becomes problematic.  Maybe @drmorr0 has thoughts?